skbio/io/format/tests/test_genbank.py

```diff
@@ -0,0 +1,429 @@
+                            'left_partial_': False,
+                            'location': '1..9',
+                            'organism': 'uncultured Xylanimonas sp.',
+                            'rc_': False,
+                            'right_partial_': False,
+                            'type_': 'source'},
+                           {'index_': 1,
+                            'left_partial_': True,
+                            'location': 'complement(<2..>8)',
+                            'product': '16S ribosomal RNA',
+                            'rc_': True,
+                            'right_partial_': True,
+                            'type_': 'rRNA'}],
+              'KEYWORDS': 'ENV.',
+              'LOCUS': {'date': datetime(2010, 8, 29, 0, 0),
+                        'division': 'ENV',
+                        'locus_name': 'HQ018078',
+                        'mol_type': 'DNA',
+                        'shape': 'linear',
+                        'size': 9,
+                        'unit': 'bp'},
+              'SOURCE': {'ORGANISM': 'uncultured Xylanimonas sp.',
+                         'taxonomy': 'Bacteria; Actinobacteria; '
+                         'Micrococcales; Promicromonosporaceae; '
+                         'Xylanimonas; environmental samples.'},
+              'VERSION': 'HQ018078.1  GI:304421728'},
+             pd.DataFrame({0: [True] * 9,
+                           1: [False] + [True] * 7 + [False]}),
+             DNA))
+
+
+class ReaderTests(GenBankIOTests):
+    def test_parse_reference(self):
+        lines = '''
+REFERENCE   1  (bases 1 to 154478)
+  AUTHORS   Sato,S., Nakamura,Y., Kaneko,T., and Tabata,S.
+  TITLE     Complete structure of the chloroplast genome of
+            Arabidopsis thaliana
+  JOURNAL   DNA Res. 6 (5), 283-290 (1999)
+   PUBMED   10574454'''.split('\n')
+
+        exp = {'AUTHORS': 'Sato,S., Nakamura,Y., Kaneko,T., and Tabata,S.',
+               'JOURNAL': 'DNA Res. 6 (5), 283-290 (1999)',
+               'PUBMED': '10574454',
+               'REFERENCE': '1  (bases 1 to 154478)',
+               'TITLE': ('Complete structure of the chloroplast genome of'
+                         ' Arabidopsis thaliana')}
+        self.assertEqual(_parse_reference(lines), exp)
+
+    def test_parse_locus(self):
+        for serialized, parsed in self.locus:
+            self.assertEqual(_parse_locus(serialized), parsed)
+
+    def test_parse_locus_invalid(self):
+        lines = [
+            # missing unit
+            ['LOCUS       NC_005816               9609 '
+             '    DNA     circular CON 07-FEB-2015'],
+            # missing division
+            ['LOCUS       SCU49845     5028 bp'
+             '    DNA                    21-JUN-1999'],
+            # wrong date format
+            ['LOCUS       NP_001832                360 aa'
+             '            linear   PRI 2001-12-18']]
+        for line in lines:
+            with self.assertRaisesRegexp(
```

Comment:
This gives a deprecation warning in `Py3` You can use [`six.assertRaisesRegex`](https://six.readthedocs.org/#six.assertRaisesRegex)